File: 0409-longest-palindrome/0409-longest-palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        int seen[128];
        for (char ch : s)
            seen[ch]++;
        bool hasCenter = false;
        int ans = 0;
        for (int freq : seen){
            ans += (freq / 2) * 2;
            if (freq % 2 != 0 && !hasCenter)
                hasCenter = true;
        }
        return hasCenter ? ans+1 : ans;
    }
};

File: 0409-longest-palindrome/0409-longest-palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        ans = 0
        count = [0] * 128
        for ch in s:
            count[ord(ch)] += 1
        hasOdd = False
        for x in count:
            ans += (x // 2) * 2
            if not hasOdd and x % 2 == 1:
                hasOdd = True
                ans += 1
        return ans
        